examples/kaleidoscope.rs

```diff
@@ -0,0 +1,1290 @@
+                let then_bb = self.context.append_basic_block(&parent, "then");
+                let else_bb = self.context.append_basic_block(&parent, "else");
+                let cont_bb = self.context.append_basic_block(&parent, "ifcont");
+
+                self.builder.build_conditional_branch(&cond, &then_bb, &else_bb);
+
+                // build then block
+                self.builder.position_at_end(&then_bb);
+                let then_val = self.compile_expr(&consequence)?;
+                self.builder.build_unconditional_branch(&cont_bb);
+
+                let then_bb = self.builder.get_insert_block().unwrap();
+
+                // build else block
+                self.builder.position_at_end(&else_bb);
+                let else_val = self.compile_expr(&alternative)?;
+                self.builder.build_unconditional_branch(&cont_bb);
+
+                let else_bb = self.builder.get_insert_block().unwrap();
+
+                // emit merge block
+                self.builder.position_at_end(&cont_bb);
+                
+                let phi = self.builder.build_phi(&self.context.f64_type(), "iftmp");
+
+                phi.add_incoming(&[
+                    (&then_val, &then_bb),
+                    (&else_val, &else_bb)
+                ]);
+
+                Ok(unsafe { std::mem::transmute(phi) })
+            },
+
+            &Expr::For { ref var_name, ref start, ref end, ref step, ref body } => {
+                let parent = self.fn_value;
+                let zero_const = self.context.f64_type().const_float(0.0);
+
+                let start = self.compile_expr(&start)?;
+                let start_alloca = self.create_entry_block_alloca(var_name, None);
+
+                self.builder.build_store(&start_alloca, &start);
+
+                // go from current block to loop block
+                let loop_bb = self.context.append_basic_block(&parent, "loop");
+
+                self.builder.build_unconditional_branch(&loop_bb);
+                self.builder.position_at_end(&loop_bb);
+
+                let variable = self.create_entry_block_alloca(var_name, None);
+
+                let old_val = match self.variables.get(var_name.as_str()) {
+                    Some(val) => Some(*val),
+                    None => None
+                };
+
+                self.variables.insert(var_name.to_owned(), variable);
+
+                self.compile_expr(&body)?;
+
+                let step = match step {
+                    &Some(ref step) => self.compile_expr(&step)?,
+                    &None => self.context.f64_type().const_float(1.0)
+                };
+
+                let end_cond = self.compile_expr(&end)?;
+
+                let curr_var = self.builder.build_load(&start_alloca, var_name);
+                let next_var = self.builder.build_float_add(&curr_var.as_float_value(), &step, "nextvar");
+
+                self.builder.build_store(&unsafe { std::mem::transmute(next_var) }, &start_alloca);
```

Comment:
Why is a float being used as a pointer? Did you mean `build_store(&start_alloca, &next_var)`?

Reply:
Good catch, thanks!